src/Linkify.jsx

```diff
@@ -55,10 +64,17 @@ class Linkify extends React.Component {
 
         props[key] = val;
       }
+      
+      let text = match.text;
+      
+      if(this.props.preCompileText){
+         text = this.props.preCompileText(match.text);
```

Comment:
I think this should be `this.props.preCompileText(match.text, props.href)` too so that you can re-write the text based on information of the URL.